tests/unit/test_table_options.py

```diff
@@ -0,0 +1,474 @@
+        setup_table(
+            faux_conn,
+            "some_table",
+            sqlalchemy.Column("createdAt", sqlalchemy.DateTime),
+            bigquery_time_partitioning=TimePartitioning(
+                field="createdAt",
+                type_="DAY",
+                expiration_ms=21600000,
+            ),
+        )
+
+    assert " ".join(faux_conn.test_data["execute"][-1][0].strip().split()) == (
+        "CREATE TABLE `some_table` ( `createdAt` DATETIME )"
+        " PARTITION BY DATE_TRUNC(createdAt, DAY)"
+        " OPTIONS(partition_expiration_days=0.25)"
+    )
+
+
+def test_table_partitioning_dialect_option_type_error(faux_conn):
+    # expect TypeError when bigquery_time_partitioning is not a TimePartitioning object
+    with pytest.raises(TypeError):
+        setup_table(
+            faux_conn,
+            "some_table",
+            sqlalchemy.Column("id", sqlalchemy.Integer),
+            sqlalchemy.Column("createdAt", sqlalchemy.DateTime),
+            bigquery_time_partitioning="DATE(createdAt)",
+        )
+
+
+def test_table_range_partitioning_dialect_option(faux_conn):
+    # expect table creation to fail as SQLite does not support partitioned tables
+    with pytest.raises(sqlite3.OperationalError):
+        setup_table(
+            faux_conn,
+            "some_table",
+            sqlalchemy.Column("id", sqlalchemy.Integer),
+            sqlalchemy.Column("zipcode", sqlalchemy.INT),
+            bigquery_range_partitioning=RangePartitioning(
+                field="zipcode",
+                range_=PartitionRange(
+                    start=0,
+                    end=100000,
+                    interval=2,
+                ),
+            ),
+        )
+
+    assert " ".join(faux_conn.test_data["execute"][-1][0].strip().split()) == (
+        "CREATE TABLE `some_table` ( `id` INT64, `zipcode` INT64 )"
+        " PARTITION BY RANGE_BUCKET(zipcode, GENERATE_ARRAY(0, 100000, 2))"
+    )
+
+
+def test_table_range_partitioning_dialect_option_no_field(faux_conn):
+    # expect TypeError when bigquery_range_partitioning field is not defined
+    with pytest.raises(
+        AttributeError,
+        match="bigquery_range_partitioning expects field to be defined",
+    ):
+        setup_table(
+            faux_conn,
+            "some_table",
+            sqlalchemy.Column("id", sqlalchemy.Integer),
+            sqlalchemy.Column("zipcode", sqlalchemy.FLOAT),
+            bigquery_range_partitioning=RangePartitioning(
+                range_=PartitionRange(
+                    start=0,
+                    end=100000,
+                    interval=10,
+                ),
+            ),
+        )
+
+
+def test_table_range_partitioning_dialect_option_bad_column_type(faux_conn):
+    # expect ValueError when bigquery_range_partitioning field is not an INTEGER
+    with pytest.raises(
+        ValueError,
+        match="bigquery_range_partitioning expects field \(i\.e\. column\) data type to be INTEGER",
```

Comment:
```suggestion
        match=r"bigquery_range_partitioning expects field \(i\.e\. column\) data type to be INTEGER",
```

Reply:
Yes, just made the change! I should have run the full test suite locally..

Reply:
I am running the test suite here. Hopefully this all works out. <fingers crossed>